### DISCUSSION TITLE
Failed to build ARDUINO_NANO_RP2040_CONNECT (mod_network_nic_type_nina_base not constant)

### DISCUSSION THREAD

**Discussion by ksprs** [micropython/micropython#10170] [RP2040 / Pico]
Hi!

It seams there is a trouble with a build process for ARDUINO_NANO_RP2040_CONNECT as it failed with these messages:

`micropython/extmod/network_ninaw10.c:831:13: error: initializer element is not constant
     .base = mod_network_nic_type_nina_base,
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/kaspars/git/micropython/extmod/network_ninaw10.c:831:13: note: (near initialization for 'mod_network_nic_type_nina.base')
make[3]: *** [CMakeFiles/firmware.dir/build.make:2624: CMakeFiles/firmware.dir/home/kaspars/git/micropython/extmod/network_ninaw10.c.obj] Error 1
make[3]: *** Waiting for unfinished jobs....
make[2]: *** [CMakeFiles/Makefile2:1383: CMakeFiles/firmware.dir/all] Error 2
make[1]: *** [Makefile:91: all] Error 2
-e See https://github.com/micropython/micropython/wiki/Build-Troubleshooting
make: *** [Makefile:33: all] Error 1`

What I did was:
`git clone https://github.com/micropython/micropython.git
cd micropython/
git submodule update --init lib/micropython-lib lib/pico-sdk lib/tinyusb lib/mbedtls lib/mynewt-nimble
cd ports/rp2/
make -j4 BOARD=ARDUINO_NANO_RP2040_CONNECT`

Am I missing something?

**Comment by jimmo** [comment]
Can you try updating to a newer GCC version?

**Comment by mattytrentini** [comment]
I ran into the same error building with buster/gcc 8.3.0. After @jimmo's suggestion, I upgraded to bullseye/gcc 10.2.1 which built fine.

**Reply by mattytrentini** [reply]
I've just updated the [micropython/build-micropython-arm](https://hub.docker.com/r/micropython/build-micropython-arm) docker build container to use Debian Bullseye - and gcc 10.2.1 - for `latest`. So this now successfully builds `ARDUINO_NANO_RP2040_CONNECT`:

```bash
> docker run -ti --rm -v $(pwd):/code -w /code micropython/build-micropython-arm bash -c "make -C mpy-cross && make -C ports/rp2 submodules all BOARD=ARDUINO_NANO_RP2040_CONNECT"
```

I also tagged the Buster variant so you can still use it (to check the build failure):

```bash
> docker run -ti --rm -v $(pwd):/code -w /code micropython/build-micropython-arm:buster bash -c "make -C mpy-cross && make -C ports/rp2 submodules all BOARD=ARDUINO_NANO_RP2040_CONNECT"
```

**Comment by ksprs** [comment]
Perfect! This solved the problem, thanks!
